src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/AzureProvider.java

```diff
@@ -599,48 +632,98 @@ protected void deleteNetworks(Azure azureService, List<com.microsoft.azure.manag
+                            .values()
+                            .stream()
+                            .filter(extension -> extension.publisherName().equals(SCRIPT_EXTENSION_PUBLISHER_WINDOWS) &&
+                                                 extension.typeName().equals(SCRIPT_EXTENSION_TYPE_WINDOWS))
+                            .findAny();
+        } else {
+            throw new RuntimeException("ERROR Operating System of type '" + vm.osType().toString() +
+                                       "' is not yet supported");
+        }
+
         if (vmExtension.isPresent()) {
-            // if the script didn't change compared to the previous update,
-            // then the script will not be run by the Azure provider.
-            // Moreover, the 'forceUpdateTag' is not available in the Azure
-            // Java SDK. So in order to make sure that the script slightly
-            // changes, we make the scripts begin with a command that prints
-            // a new ID each time a script needs to be run after the first
-            // executed script.
-            UUID scriptExecutionId = UUID.randomUUID();
-            String concatenatedScriptsWithExecutionId = concatenatedScripts.insert(0,
-                                                                                   "echo script-ID" + SCRIPT_SEPARATOR +
-                                                                                      "echo " + scriptExecutionId +
-                                                                                      SCRIPT_SEPARATOR)
-                                                                           .toString();
-            log.info("Request Azure provider to execute script: " + concatenatedScriptsWithExecutionId);
-            vm.update()
-              .updateExtension(vmExtension.get().name())
-              .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY, concatenatedScriptsWithExecutionId)
-              .parent()
-              .apply();
+            log.info("Request Azure provider to execute script: " + concatenatedScripts);
+            AbstractFuture<VirtualMachine> vmfuture = vm.update()
+                                                        .updateExtension(vmExtension.get().name())
+                                                        .withPublicSetting(SCRIPT_EXTENSION_CMD_KEY,
+                                                                           concatenatedScripts.toString())
+                                                        .parent()
+                                                        .applyAsync(null);
+            //.addListener(Runnables.doNothing(), Executors.newSingleThreadExecutor());
+            try {
+                vmfuture.get(TIMEOUT_SCRIPT_EXECUTION, TimeUnit.MINUTES);
+            } catch (InterruptedException e) {
+                log.info("An interruption occurred when updating extension to execute script on a VM.");
+                e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1148)